### ISSUE TITLE
Spring Boot Logging Configuration 

### ISSUE THREAD

**Issue by chhex** [spring-projects/spring-boot#16851]

**Comment by wilkinsona** [comment]
I think we might be reinventing the wheel a bit if we did anything here. For example, Logback will provide debug information about its configuration when the JVM is started with `Dlogback.debug=true`. Does that might your needs?

**Comment by chhex** [comment]
Background:  I have seen Spring Boot Applications, which are configured the same way at least in terms of Spring Logging, but behave completely different, because of dependencies in the classpath. Sometimes it is even third party libraries, which "break" the logging contract. I should be possible to debug these situations more easy. 

In `org.springframework.boot.logging.LoggingApplicationListener` the only Logging Output is in case of Exceptions.  

Something like -Dlogback.debug=true would be great

**Comment by chhex** [comment]
I had situations, where -Dlogback.debug=true didn't help.

**Comment by philwebb** [comment]
> Unfortunately one get through many third libraries logging libraries as transitive dependency, which can lead to curious situations in terms of logging configuration

I'm curious to know which libraries caused problems and how they interfered with Spring Boot's logging setup. The logging situation in Java is certainly complex and I know it's pretty difficult to always get all the various libraries to play nice.

> it would be great to have for example after Spring Boot Startup the possibility to log the Logging configuration, which Spring Boot uses.

I'm not 100% sure this will be possible since the only real source of truth is the logging library itself. It's already possible to configure logback using Spring Boot, or directly with XML. It's also possible to configure it programmatically. I'm not sure how we can present the logging configuration in a more meaningful way than logback could itself.

I think it would be useful if we could have some examples of the types of output you'd find useful. I'm always keen to add useful more diagnostics features if we can. If you have time to put together a sample where the logging fails in an unexpected way, that would also be very useful.

**Comment by chhex** [comment]
It's clear that spring-boot-logging is "only" a layer above the divers Java Logging System, it supports, basically with the goal to facilitate there configuration. 

My concrete scenario: 
I got a spring boot project from other developers, which runs fine in the ide & builds with testcases. My job is basically to Integrate to application into a delivery pipeline, the delivery being a rpm.  The contracts basically the runtime dependencies of the application. We have a standardized rpm "template" using the nebula rpm plugin. Nothing special here.  We configure the logging with spring boot application properties of the following form: 

`logging.file=log/application-service-targetenv.log
logging.config=classpath:app/logback.xml`

The logback configuration is pretty much straight from Spring Boot documentation:
`configuration>

	<include resource="org/springframework/boot/logging/logback/defaults.xml" />
	<jmxConfigurator/>
	<property name="LOG_FILE"
		value="${LOG_FILE:-${LOG_PATH:-${LOG_TEMP:-${java.io.tmpdir:-/tmp}}/}spring.log}" />
	<include resource="org/springframework/boot/logging/logback/file-appender.xml" />
	<include resource="org/springframework/boot/logging/logback/console-appender.xml" />
		
<root level="INFO">
		<appender-ref ref="FILE" />
</root>	
</configuration>`
and reuses the Spring Boot appender configurations.

That works mostly great.   Every once in a while we get applications , which breaks the configuration. Various reasons: Multiple Logging Implementation Libraries , missing Logging Bridges, Missing Implementation Libraries, own appender configurations, which don't work as expected etc. Business as usual. 

This project had in the end two problems:

1. log4j and logback libraries
2. This is a new one for me: It had transitively a unneeded spring-cloud-starter dependency.  

The error symptom was that logback didn't log as expected to the fileappender, but the console appender. 

This one was quite a surprise for me and a new one, which i found out basically reducing the transitive dependencies, building , running tests and deploying and testing again.  I don't know how the spring-cloud dependency screwed up the spring-boot-logging.  I only know , that when i got rid of the dependency, everything was back to the expected again. 

Naturally, we try to reduced the variability in the applications, by having standardized poms , starters und using the spring boot dependency management. But there is still enough (accidentional) variability, which sometimes may lead to weird situations, like this one. 

During my "debugging" sessions,  i came up with this issue.  It's always great to have information in the scenario , where the things don't  work as expected and mostly it's there. Interesting enough Java Logging seems to be one of those "hotspots" of producing unexpected behavior.

**Comment by philwebb** [comment]
Thanks for the additional background, that's really useful. There's certainly more we can do in this area I think. I tried a few applications to see what the standard logback debug output looks like and got some interesting results.

If you start `SampleTomcatApplication` from our repo with `-Dlogback.debug=true` you get no debug output at all. I think this is because we're programmatically configuring logback so `ch.qos.logback.classic.joran.action.ConfigurationAction` doesn't run.

If you start `SampleActuatorApplication` which has a logback.xml file you get:

```
11:57:10,453 |-INFO in ch.qos.logback.classic.LoggerContext[default] - Could NOT find resource [logback-test.xml]
11:57:10,453 |-INFO in ch.qos.logback.classic.LoggerContext[default] - Could NOT find resource [logback.groovy]
11:57:10,453 |-INFO in ch.qos.logback.classic.LoggerContext[default] - Found resource [logback.xml] at [file:/Users/pwebb/projects/spring-boot/code21x/spring-boot-samples/spring-boot-sample-actuator/target/classes/logback.xml]
11:57:10,525 |-INFO in ch.qos.logback.core.joran.util.ConfigurationWatchListUtil@277c0f21 - Adding [file:/Users/pwebb/projects/spring-boot/code21x/spring-boot-project/spring-boot/target/classes/org/springframework/boot/logging/logback/base.xml] to configuration watch list.
11:57:10,527 |-INFO in ch.qos.logback.core.joran.util.ConfigurationWatchListUtil@277c0f21 - Adding [file:/Users/pwebb/projects/spring-boot/code21x/spring-boot-project/spring-boot/target/classes/org/springframework/boot/logging/logback/defaults.xml] to configuration watch list.
11:57:10,529 |-INFO in ch.qos.logback.core.joran.action.ConversionRuleAction - registering conversion word clr with class [org.springframework.boot.logging.logback.ColorConverter]
11:57:10,529 |-INFO in ch.qos.logback.core.joran.action.ConversionRuleAction - registering conversion word wex with class [org.springframework.boot.logging.logback.WhitespaceThrowableProxyConverter]
11:57:10,529 |-INFO in ch.qos.logback.core.joran.action.ConversionRuleAction - registering conversion word wEx with class [org.springframework.boot.logging.logback.ExtendedWhitespaceThrowableProxyConverter]
11:57:10,532 |-INFO in ch.qos.logback.classic.joran.action.LoggerAction - Setting level of logger [org.apache.catalina.startup.DigesterFactory] to ERROR
11:57:10,533 |-INFO in ch.qos.logback.classic.joran.action.LoggerAction - Setting level of logger [org.apache.catalina.util.LifecycleBase] to ERROR
11:57:10,533 |-INFO in ch.qos.logback.classic.joran.action.LoggerAction - Setting level of logger [org.apache.coyote.http11.Http11NioProtocol] to WARN
11:57:10,533 |-INFO in ch.qos.logback.classic.joran.action.LoggerAction - Setting level of logger [org.apache.sshd.common.util.SecurityUtils] to WARN
11:57:10,533 |-INFO in ch.qos.logback.classic.joran.action.LoggerAction - Setting level of logger [org.apache.tomcat.util.net.NioSelectorPool] to WARN
11:57:10,533 |-INFO in ch.qos.logback.classic.joran.action.LoggerAction - Setting level of logger [org.eclipse.jetty.util.component.AbstractLifeCycle] to ERROR
11:57:10,533 |-INFO in ch.qos.logback.classic.joran.action.LoggerAction - Setting level of logger [org.hibernate.validator.internal.util.Version] to WARN
11:57:10,533 |-INFO in ch.qos.logback.core.joran.util.ConfigurationWatchListUtil@277c0f21 - Adding [file:/Users/pwebb/projects/spring-boot/code21x/spring-boot-project/spring-boot/target/classes/org/springframework/boot/logging/logback/console-appender.xml] to configuration watch list.
11:57:10,535 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - About to instantiate appender of type [ch.qos.logback.core.ConsoleAppender]
11:57:10,538 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - Naming appender as [CONSOLE]
11:57:10,545 |-INFO in ch.qos.logback.core.joran.action.NestedComplexPropertyIA - Assuming default type [ch.qos.logback.classic.encoder.PatternLayoutEncoder] for [encoder] property
11:57:10,569 |-INFO in ch.qos.logback.core.joran.util.ConfigurationWatchListUtil@277c0f21 - Adding [file:/Users/pwebb/projects/spring-boot/code21x/spring-boot-project/spring-boot/target/classes/org/springframework/boot/logging/logback/file-appender.xml] to configuration watch list.
11:57:10,571 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - About to instantiate appender of type [ch.qos.logback.core.rolling.RollingFileAppender]
11:57:10,574 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - Naming appender as [FILE]
11:57:10,575 |-INFO in ch.qos.logback.core.joran.action.NestedComplexPropertyIA - Assuming default type [ch.qos.logback.classic.encoder.PatternLayoutEncoder] for [encoder] property
11:57:10,583 |-INFO in c.q.l.core.rolling.SizeAndTimeBasedRollingPolicy@873415566 - Archive files will be limited to [10 MB] each.
11:57:10,586 |-INFO in c.q.l.core.rolling.SizeAndTimeBasedRollingPolicy@873415566 - Will use gz compression
11:57:10,587 |-INFO in c.q.l.core.rolling.SizeAndTimeBasedRollingPolicy@873415566 - Will use the pattern /var/folders/w5/8bwfh56d7bq2jw678lvns08r0000gn/T//spring.log.%d{yyyy-MM-dd}.%i for the active file
11:57:10,589 |-INFO in ch.qos.logback.core.rolling.SizeAndTimeBasedFNATP@35f983a6 - The date pattern is 'yyyy-MM-dd' from file name pattern '/var/folders/w5/8bwfh56d7bq2jw678lvns08r0000gn/T//spring.log.%d{yyyy-MM-dd}.%i.gz'.
11:57:10,590 |-INFO in ch.qos.logback.core.rolling.SizeAndTimeBasedFNATP@35f983a6 - Roll-over at midnight.
11:57:10,593 |-INFO in ch.qos.logback.core.rolling.SizeAndTimeBasedFNATP@35f983a6 - Setting initial period to Wed May 15 11:56:51 PDT 2019
11:57:10,624 |-INFO in ch.qos.logback.core.rolling.RollingFileAppender[FILE] - Active log file name: /var/folders/w5/8bwfh56d7bq2jw678lvns08r0000gn/T//spring.log
11:57:10,624 |-INFO in ch.qos.logback.core.rolling.RollingFileAppender[FILE] - File property is set to [/var/folders/w5/8bwfh56d7bq2jw678lvns08r0000gn/T//spring.log]
11:57:10,625 |-INFO in ch.qos.logback.classic.joran.action.RootLoggerAction - Setting level of ROOT logger to INFO
11:57:10,625 |-INFO in ch.qos.logback.core.joran.action.AppenderRefAction - Attaching appender named [CONSOLE] to Logger[ROOT]
11:57:10,626 |-INFO in ch.qos.logback.core.joran.action.AppenderRefAction - Attaching appender named [FILE] to Logger[ROOT]
11:57:10,626 |-INFO in ch.qos.logback.classic.joran.action.RootLoggerAction - Setting level of ROOT logger to INFO
11:57:10,626 |-INFO in ch.qos.logback.core.joran.action.AppenderRefAction - Attaching appender named [CONSOLE] to Logger[ROOT]
11:57:10,626 |-INFO in ch.qos.logback.classic.joran.action.ConfigurationAction - End of configuration.
11:57:10,627 |-INFO in ch.qos.logback.classic.joran.JoranConfigurator@7a5d012c - Registering current configuration as safe fallback point
11:57:10,935 |-INFO in ch.qos.logback.core.joran.util.ConfigurationWatchListUtil@277c0f21 - Adding [file:/Users/pwebb/projects/spring-boot/code21x/spring-boot-project/spring-boot/target/classes/org/springframework/boot/logging/logback/base.xml] to configuration watch list.
11:57:10,936 |-INFO in ch.qos.logback.core.joran.util.ConfigurationWatchListUtil@277c0f21 - Adding [file:/Users/pwebb/projects/spring-boot/code21x/spring-boot-project/spring-boot/target/classes/org/springframework/boot/logging/logback/defaults.xml] to configuration watch list.
11:57:10,937 |-INFO in ch.qos.logback.core.joran.action.ConversionRuleAction - registering conversion word clr with class [org.springframework.boot.logging.logback.ColorConverter]
11:57:10,937 |-INFO in ch.qos.logback.core.joran.action.ConversionRuleAction - registering conversion word wex with class [org.springframework.boot.logging.logback.WhitespaceThrowableProxyConverter]
11:57:10,937 |-INFO in ch.qos.logback.core.joran.action.ConversionRuleAction - registering conversion word wEx with class [org.springframework.boot.logging.logback.ExtendedWhitespaceThrowableProxyConverter]
11:57:10,938 |-INFO in ch.qos.logback.classic.joran.action.LoggerAction - Setting level of logger [org.apache.catalina.startup.DigesterFactory] to ERROR
11:57:10,938 |-INFO in ch.qos.logback.classic.jul.LevelChangePropagator@71a794e5 - Propagating ERROR level on Logger[org.apache.catalina.startup.DigesterFactory] onto the JUL framework
11:57:10,938 |-INFO in ch.qos.logback.classic.joran.action.LoggerAction - Setting level of logger [org.apache.catalina.util.LifecycleBase] to ERROR
11:57:10,939 |-INFO in ch.qos.logback.classic.jul.LevelChangePropagator@71a794e5 - Propagating ERROR level on Logger[org.apache.catalina.util.LifecycleBase] onto the JUL framework
11:57:10,940 |-INFO in ch.qos.logback.classic.joran.action.LoggerAction - Setting level of logger [org.apache.coyote.http11.Http11NioProtocol] to WARN
11:57:10,940 |-INFO in ch.qos.logback.classic.jul.LevelChangePropagator@71a794e5 - Propagating WARN level on Logger[org.apache.coyote.http11.Http11NioProtocol] onto the JUL framework
11:57:10,940 |-INFO in ch.qos.logback.classic.joran.action.LoggerAction - Setting level of logger [org.apache.sshd.common.util.SecurityUtils] to WARN
11:57:10,940 |-INFO in ch.qos.logback.classic.jul.LevelChangePropagator@71a794e5 - Propagating WARN level on Logger[org.apache.sshd.common.util.SecurityUtils] onto the JUL framework
11:57:10,941 |-INFO in ch.qos.logback.classic.joran.action.LoggerAction - Setting level of logger [org.apache.tomcat.util.net.NioSelectorPool] to WARN
11:57:10,941 |-INFO in ch.qos.logback.classic.jul.LevelChangePropagator@71a794e5 - Propagating WARN level on Logger[org.apache.tomcat.util.net.NioSelectorPool] onto the JUL framework
11:57:10,941 |-INFO in ch.qos.logback.classic.joran.action.LoggerAction - Setting level of logger [org.eclipse.jetty.util.component.AbstractLifeCycle] to ERROR
11:57:10,941 |-INFO in ch.qos.logback.classic.jul.LevelChangePropagator@71a794e5 - Propagating ERROR level on Logger[org.eclipse.jetty.util.component.AbstractLifeCycle] onto the JUL framework
11:57:10,941 |-INFO in ch.qos.logback.classic.joran.action.LoggerAction - Setting level of logger [org.hibernate.validator.internal.util.Version] to WARN
11:57:10,941 |-INFO in ch.qos.logback.classic.jul.LevelChangePropagator@71a794e5 - Propagating WARN level on Logger[org.hibernate.validator.internal.util.Version] onto the JUL framework
11:57:10,941 |-INFO in ch.qos.logback.core.joran.util.ConfigurationWatchListUtil@277c0f21 - Adding [file:/Users/pwebb/projects/spring-boot/code21x/spring-boot-project/spring-boot/target/classes/org/springframework/boot/logging/logback/console-appender.xml] to configuration watch list.
11:57:10,942 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - About to instantiate appender of type [ch.qos.logback.core.ConsoleAppender]
11:57:10,942 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - Naming appender as [CONSOLE]
11:57:10,942 |-INFO in ch.qos.logback.core.joran.action.NestedComplexPropertyIA - Assuming default type [ch.qos.logback.classic.encoder.PatternLayoutEncoder] for [encoder] property
11:57:10,943 |-INFO in ch.qos.logback.core.joran.util.ConfigurationWatchListUtil@277c0f21 - Adding [file:/Users/pwebb/projects/spring-boot/code21x/spring-boot-project/spring-boot/target/classes/org/springframework/boot/logging/logback/file-appender.xml] to configuration watch list.
11:57:10,945 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - About to instantiate appender of type [ch.qos.logback.core.rolling.RollingFileAppender]
11:57:10,945 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - Naming appender as [FILE]
11:57:10,945 |-INFO in ch.qos.logback.core.joran.action.NestedComplexPropertyIA - Assuming default type [ch.qos.logback.classic.encoder.PatternLayoutEncoder] for [encoder] property
11:57:10,946 |-INFO in c.q.l.core.rolling.SizeAndTimeBasedRollingPolicy@1073533248 - Archive files will be limited to [10 MB] each.
11:57:10,946 |-INFO in c.q.l.core.rolling.SizeAndTimeBasedRollingPolicy@1073533248 - Will use gz compression
11:57:10,946 |-INFO in c.q.l.core.rolling.SizeAndTimeBasedRollingPolicy@1073533248 - Will use the pattern /var/folders/w5/8bwfh56d7bq2jw678lvns08r0000gn/T//spring.log.%d{yyyy-MM-dd}.%i for the active file
11:57:10,947 |-INFO in ch.qos.logback.core.rolling.SizeAndTimeBasedFNATP@23bb8443 - The date pattern is 'yyyy-MM-dd' from file name pattern '/var/folders/w5/8bwfh56d7bq2jw678lvns08r0000gn/T//spring.log.%d{yyyy-MM-dd}.%i.gz'.
11:57:10,947 |-INFO in ch.qos.logback.core.rolling.SizeAndTimeBasedFNATP@23bb8443 - Roll-over at midnight.
11:57:10,948 |-INFO in ch.qos.logback.core.rolling.SizeAndTimeBasedFNATP@23bb8443 - Setting initial period to Wed May 15 11:56:51 PDT 2019
11:57:10,970 |-INFO in ch.qos.logback.core.rolling.RollingFileAppender[FILE] - Active log file name: /var/folders/w5/8bwfh56d7bq2jw678lvns08r0000gn/T//spring.log
11:57:10,970 |-INFO in ch.qos.logback.core.rolling.RollingFileAppender[FILE] - File property is set to [/var/folders/w5/8bwfh56d7bq2jw678lvns08r0000gn/T//spring.log]
11:57:10,971 |-INFO in ch.qos.logback.classic.joran.action.RootLoggerAction - Setting level of ROOT logger to INFO
11:57:10,971 |-INFO in ch.qos.logback.classic.jul.LevelChangePropagator@71a794e5 - Propagating INFO level on Logger[ROOT] onto the JUL framework
11:57:10,971 |-INFO in ch.qos.logback.core.joran.action.AppenderRefAction - Attaching appender named [CONSOLE] to Logger[ROOT]
11:57:10,971 |-INFO in ch.qos.logback.core.joran.action.AppenderRefAction - Attaching appender named [FILE] to Logger[ROOT]
11:57:10,971 |-INFO in ch.qos.logback.classic.joran.action.RootLoggerAction - Setting level of ROOT logger to INFO
11:57:10,971 |-INFO in ch.qos.logback.core.joran.action.AppenderRefAction - Attaching appender named [CONSOLE] to Logger[ROOT]
11:57:10,971 |-INFO in ch.qos.logback.classic.joran.action.ConfigurationAction - End of configuration.
11:57:10,971 |-INFO in org.springframework.boot.logging.logback.SpringBootJoranConfigurator@120d6fe6 - Registering current configuration as safe fallback point

  .   ____          _            __ _ _
 /\\ / ___'_ __ _ _(_)_ __  __ _ \ \ \ \
( ( )\___ | '_ | '_| | '_ \/ _` | \ \ \ \
 \\/  ___)| |_)| | | | | || (_| |  ) ) ) )
  '  |____| .__|_| |_|_| |_\__, | / / / /
 =========|_|==============|___/=/_/_/_/
 :: Spring Boot ::                        

2019-05-15 11:57:11.078  INFO 74205 --- [           main] s.actuator.SampleActuatorApplication     : Starting SampleActuatorApplication on pwmb16 with PID 74205 (/Users/pwebb/projects/spring-boot/code21x/spring-boot-samples/spring-boot-sample-actuator/target/classes started by pwebb in /Users/pwebb/projects/spring-boot/code21x/spring-boot-samples/spring-boot-sample-actuator)
2019-05-15 11:57:11.080  INFO 74205 --- [           main] s.actuator.SampleActuatorApplication     : No active profile set, falling back to default profiles: default
2019-05-15 11:57:12.277  INFO 74205 --- [           main] trationDelegate$BeanPostProcessorChecker : Bean 'org.springframework.transaction.annotation.ProxyTransactionManagementConfiguration' of type [org.springframework.transaction.annotation.ProxyTransactionManagementConfiguration$$EnhancerBySpringCGLIB$$123ea0a] is not eligible for getting processed by all BeanPostProcessors (for example: not eligible for auto-proxying)
2019-05-15 11:57:12.557  INFO 74205 --- [           main] o.s.b.w.embedded.tomcat.TomcatWebServer  : Tomcat initialized with port(s): 8080 (http)
2019-05-15 11:57:12.583  INFO 74205 --- [           main] o.apache.catalina.core.StandardService   : Starting service [Tomcat]
2019-05-15 11:57:12.583  INFO 74205 --- [           main] org.apache.catalina.core.StandardEngine  : Starting Servlet engine: [Apache Tomcat/9.0.19]
2019-05-15 11:57:12.677  INFO 74205 --- [           main] o.a.c.c.C.[Tomcat].[localhost].[/]       : Initializing Spring embedded WebApplicationContext
2019-05-15 11:57:12.679  INFO 74205 --- [           main] o.s.web.context.ContextLoader            : Root WebApplicationContext: initialization completed in 1556 ms
2019-05-15 11:57:13.270  INFO 74205 --- [           main] o.s.s.concurrent.ThreadPoolTaskExecutor  : Initializing ExecutorService 'applicationTaskExecutor'
2019-05-15 11:57:13.741  INFO 74205 --- [           main] o.s.s.web.DefaultSecurityFilterChain     : Creating filter chain: any request, [org.springframework.security.web.context.request.async.WebAsyncManagerIntegrationFilter@5ac6c4f2, org.springframework.security.web.context.SecurityContextPersistenceFilter@205bed61, org.springframework.security.web.header.HeaderWriterFilter@4088741b, org.springframework.security.web.csrf.CsrfFilter@1c6c6f24, org.springframework.security.web.authentication.logout.LogoutFilter@204e90f7, org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter@7f08caf, org.springframework.security.web.authentication.ui.DefaultLoginPageGeneratingFilter@31d6f3fe, org.springframework.security.web.authentication.ui.DefaultLogoutPageGeneratingFilter@2aa6311a, org.springframework.security.web.authentication.www.BasicAuthenticationFilter@54657dd2, org.springframework.security.web.savedrequest.RequestCacheAwareFilter@23592946, org.springframework.security.web.servletapi.SecurityContextHolderAwareRequestFilter@10fbbdb, org.springframework.security.web.authentication.AnonymousAuthenticationFilter@61f39bb, org.springframework.security.web.session.SessionManagementFilter@54bca971, org.springframework.security.web.access.ExceptionTranslationFilter@71466383, org.springframework.security.web.access.intercept.FilterSecurityInterceptor@5fac521d]
2019-05-15 11:57:13.782  INFO 74205 --- [           main] o.s.b.a.e.web.EndpointLinksResolver      : Exposing 16 endpoint(s) beneath base path '/actuator'
2019-05-15 11:57:13.889  INFO 74205 --- [           main] o.s.b.w.embedded.tomcat.TomcatWebServer  : Tomcat started on port(s): 8080 (http) with context path ''
2019-05-15 11:57:13.893  INFO 74205 --- [           main] s.actuator.SampleActuatorApplication     : Started SampleActuatorApplication in 3.145 seconds (JVM running for 3.606)
```

There's a lot of information in there, including the files that have been loaded. Unfortunately, it's also quite hard to parse.

I quite like the idea of using your `EventListener` trick to report some minimal logging information. I think we should also try to make sure that `-Dlogback.debug=true` works with a standard boot app.

**Comment by philwebb** [comment]
I've raised #16876 to address the debug out issue. We'll leave this issue for considering what additional diagnostics we could offer.
